lambdaclass/mirra_backend
# [GH-322] Implement untargetable tag

## Motivation

We want some skills to leave their targets as Untargetables. That is, no other unit will be able to use their skills on them.

Closes https://github.com/lambdaclass/champions_of_mirra/issues/322

## Summary of changes

Before choosing the targets of a skill, filter the units that have the Untargetable tag.

## How to test it?

The easiest way to test this PR is to follow the steps in the test that is added, where a unit is always marked as untargetable and is never damaged since it receives the tag.

To test with the client, you can follow these steps:
1. Start the client and choose one of your units
2. Edit the `skills.json` file adding a tag component to one of your unit's skills. You should add something like this to an effect's `components` list
```
{
    "type": "ApplyTags",
    "tags": [
      "Untargetable"
    ]
}
```
3. Run `Champions.Config.import_skill_config()` in your terminal to update your unit's skills
4. Run a battle and check the log. You should see how the Untargetable tag is applied each time the skill you edited is used. Remember to change all Logger.info for IO.inspect to have the complete log.

## Checklist
